R/slide.R

```diff
@@ -363,3 +364,403 @@ epi_slide <- function(x, f, ..., before, after, ref_time_values,
 
   return(x)
 }
+
+#' Optimized slide function for performing rolling averages on an `epi_df` object
+#'
+#' Slides a n-timestep mean over variables in an `epi_df` object. See the [slide
+#' vignette](https://cmu-delphi.github.io/epiprocess/articles/slide.html) for
+#' examples.
+#'
+#' @param x The `epi_df` object under consideration, [grouped][dplyr::group_by]
+#'   or ungrouped. If ungrouped, all data in `x` will be treated as part of a
+#'   single data group.
+#' @param col_name A character vector of the names of one or more columns for
```

Comment:
Hm, it could be nice to support that. Since I'm already using `...` to pass extra args to the slide function (e.g. `frollmean`), and I'd like this function to be able to slide over multiple columns at once, we'd need to pass tidyselect cols as a vector. Seems like this makes output col naming more difficult.

Another thought: manual data mask handling in either `epix_slide` or `epi_slide` turned out to be quite slow. It would be undesirable if we have to do that here to support tidyselect, given the goal of making this function fast.

Reply:
Switched to tidyselection in https://github.com/cmu-delphi/epiprocess/pull/400/commits/81386cb301b2f0a016d4d1b950ffb4e91e8c8d1e. It is ~20% slower compared to using a character vector.

Reply:
Re-sped up in https://github.com/cmu-delphi/epiprocess/pull/433